### PR TITLE
feat(wasm): Reenable wasm backend

### PR DESCRIPTION
**Pull request by kobyhallx**
Need to re-enable wasm backend to publish backend

### CODE REVIEW

File: aztec_backend_wasm/src/lib.rs [hunk 1]
```diff
@@ -21,7 +23,7 @@ pub fn compute_witnesses(
 ) -> ComputedWitness {
     console_error_panic_hook::set_once();
 
-    let circuit: Circuit = circuit.into_serde().unwrap();
+    let circuit: Circuit = JsValueSerdeExt::into_serde(&circuit).unwrap();
```
Comment (jfecher) [inline aztec_backend_wasm/src/lib.rs:26]: Why the change from method to function syntax? Is there another `into_serde` on the circuit type?

File: aztec_backend_wasm/src/lib.rs [hunk 2]
```diff
@@ -42,32 +44,46 @@ pub fn compute_witnesses(
     // Now use the partial witness generator to fill in the rest of the witnesses
     // which are possible
 
+    use barretenberg_wasm::Plonk;
     let plonk = Plonk;
+    let num_witnesses = circuit.num_vars();
```
Comment (jfecher) [inline aztec_backend_wasm/src/lib.rs:49]: Nitpick: I don't think we need the `use` when the only line that uses Plonk is line 48

File: aztec_backend_wasm/src/lib.rs (hunk 1 above)
Comment (TomAFrench) [reply aztec_backend_wasm/src/lib.rs:26]: This is matching a change we performed in https://github.com/noir-lang/noir/pull/605

tl;dr: there's a circular dependency in `wasm-bindgen` which is resolved by doing this.

File: Cargo.toml [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
-members = ["common", "barretenberg_wasm", "barretenberg_static_lib"]
 
-exclude = ["aztec_backend_wasm"]
+members = ["common", "barretenberg_wasm", "barretenberg_static_lib", "aztec_backend_wasm"]
+# barretenberg_js will be phased out in the future
 
-[workspace.package]
+# exclude = ["aztec_backend_wasm"]
+
+exclude = ["barretenberg_js"]
```
Comment (TomAFrench) [inline Cargo.toml:9]: References to `barretenberg_js` are stale.

File: Cargo.toml [hunk 4]
```diff
@@ -1,11 +1,14 @@
 [workspace]
 
-members = ["common", "barretenberg_wasm", "barretenberg_static_lib"]
 
-exclude = ["aztec_backend_wasm"]
+members = ["common", "barretenberg_wasm", "barretenberg_static_lib", "aztec_backend_wasm"]
+# barretenberg_js will be phased out in the future
 
-[workspace.package]
+# exclude = ["aztec_backend_wasm"]
```
Comment (TomAFrench) [inline Cargo.toml:7]: Can delete this commented line as well.

File: aztec_backend_wasm/src/lib.rs [hunk 5]
```diff
@@ -5,6 +5,8 @@ pub use barretenberg_wasm::Barretenberg;
 
 use wasm_bindgen::prelude::*;
 
+use gloo_utils::format::JsValueSerdeExt;
```
Comment (TomAFrench) [inline aztec_backend_wasm/src/lib.rs:8]: Can we add `gloo` to cspell?

File: aztec_backend_wasm/src/lib.rs [hunk 6]
```diff
@@ -42,32 +44,46 @@ pub fn compute_witnesses(
...
+        let value = match witness_map.get(&Witness(i)) {
+            Some(value) => *value,
+            None => FieldElement::zero(),
+        };
+
+        sorted_witness.push(value);
+    }
 
-    assignments.to_bytes()
+    sorted_witness.to_bytes()
```
Comment (TomAFrench) [inline aztec_backend_wasm/src/lib.rs:69]: This change needs to be reverted. This logic has been factored out into `proof::flatten_witness_map`

File: barretenberg_wasm/src/lib.rs [hunk 7]
```diff
@@ -15,7 +15,6 @@ pub mod scalar_mul;
 pub mod schnorr;
 
 pub use common::crs;
-use std::cell::Cell;
```
Comment (TomAFrench) [inline barretenberg_wasm/src/lib.rs:18]: We make use of this on L80. Shouldn't we make that a qualified usage of `Cell` if we're removing this import?

**Review by TomAFrench — CHANGES_REQUESTED** [review]

**Comment by phated** [comment]
I believe this was superseded by #94
